tests/conftest.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+from hypothesis.extra.numpy import arrays, from_dtype
+
+
+def normalize_quaternion(quaternion):
+    if np.linalg.norm(quaternion < 1e-16):
+        return np.array((0, 0, 0, 1))
+
+    return quaternion / np.linalg.norm(quaternion)
```

Comment:
Can you put these in the functional API?

Reply:
Of course. I mainly wrote this to generate quaternions for testing via hypothesis, but we could turn this into a function that is available in the public API if you find it useful.

Reply:
I think you proved it's usefulness already :)

Reply:
Okay. In this case, I would still keep a copy in `conftest.py`. The idea during a unit-test is to not trust the library you are testing, so it feels strange to use functions from that exact library to test the libraries behavior.

Reply:
Well. Think of the functional API as a separate package with its own tests (you'd have to add these). You are consuming that from the object API.